### ISSUE TITLE
link profile.csv columns with output asc files

### ISSUE THREAD

**Issue by landam** [storm-fsv-cvut/smoderp2d#88]
profile.csv:

* maximalSurfaceFlow[m3/s] -> mQsur_m3_s.asc (?)
* totalRunoff[m3] -> ?
* maximalSurfaceRunoffVelocity[m/s] -> mVel_m_s.asc (?)
* maximalTangentialStress[Pa] -> mrSearStr_Pa.asc (?)
* rillRunoff[Y/N] -> ?

**Comment by landam** [comment]
Please review @kavkapet @jerabekjak

**Comment by pesekon2** [comment]
Output: 

```
tests/data/output/control:
cRillVOut_m3.asc  CumVRestL3.asc  mQRill_m3_s.asc   mWLevel_m.asc      surRet_M.asc
cSheetVOutM3.asc  cVIn_M3.asc     mQsheet_m3_s.asc  mWLevelRill_m.asc  widthRill.asc

tests/data/output/core:
cInfil_m3.asc  cVsur_m3.asc     mQsur_m3_s.asc    mVel_m_s.asc  volRest_m3.asc
cRain_m3.asc   massBalance.asc  mrSearStr_Pa.asc  reachFid.asc

tests/data/output/temp:
hcrit_flux.asc  hcrit_tau.asc  hcrit_v.asc  mat_nan.asc
```

**Comment by pesekon2** [comment]
Output profile:

```
length[m],soil_vegFID,maximalSurfaceFlow[m3/s],totalRunoff[m3],maximalSurfaceRunoffVelocity[m/s],maximalTangentialStress[Pa],rillRunoff[Y/N]
1,HH_pvc,0.0100,2.000,0.4500,12.10,0
2,HH_pvc,0.0105,2.100,0.4725,12.71,0
3,HH_pvc,0.0110,2.205,0.4961,13.34,0
```

**Comment by pesekon2** [comment]
Most of that solved in commit 48a1db6d382194a28519249b9998ee002a21e5f7. I hope that I haven't misinterpreted anything. 

The only one missing is `rillRunoff[Y/N]`. How do we decide whether it should be 1 or 0?

**Comment by pesekon2** [comment]
Now also `rillRunoff[Y/N]` added - after a discussion with @jerabekjak, `surface_array.state` was chosen to be put there. 

However, with the test data, I reach either values `0` or `2`. Is that expected? Shouldn't it be `0` and `1`?

**Comment by jerabekjak** [comment]
The state can reach values 0, 1, or 2. 
- 0 -> no rill flow -> rillRunoff[Y/N] = N
- 1 -> rill flow is present and the rills size increase -> rillRunoff[Y/N] = Y
- 2 -> rill flow is present and the rills size is constant -> rillRunoff[Y/N] = Y

**Comment by pesekon2** [comment]
Thanks for the explanation. Implemented it in 8afb0aa to write `0` to `rillRunoff` when the `state` is `0`, otherwise `1`. 

Closing the issue as it seems like we're finally writing everything.
